sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -783,20 +783,23 @@ object GpuCast extends Arm {
         else {strCol.incRefCount}
       }
 
-      withResource(ColumnVector.fromScalar(sep, numRows)) {sepCol =>
-        withResource(input.getChildColumnView(0)) { childCol =>
-          withResource(addSpaces(childCol)) {strChildCol =>
-            withResource(input.replaceListChild(strChildCol)) {strArrayCol =>
-              withResource(
-                strArrayCol.stringConcatenateListElements(sepCol)) { strColContainsNull =>
-                withResource(strColContainsNull.replaceNulls(empty)) {strCol =>
-                  removeFirstSpace(strCol)
-                }
-              }
-            }
-          }
+      val strChildCol = withResource(input.getChildColumnView(0)) {
+        addSpaces
+      }
+      val strArrayCol = withResource(strChildCol) {
+        input.replaceListChild
+      }
+      val strColContainsNull = withResource(ColumnVector.fromScalar(sep, numRows)) { sepCol =>
```

Comment:
If `ColumnVector.fromScalar` throws then we leak `strArrayCol` here.

Reply:
thanks, that got ordered not as I intended